static/js/dashboard.js

```diff
@@ -11,280 +12,485 @@ const dateRange = (startDate, endDate) => {
+                            <li>
+                                <form action="/api/tasks/${task.id}" onsubmit="deleteTask(event)">
+                                    <button type="submit"
+                                        class="flex items-center w-full px-4 py-2 text-red-600 hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white">
+                                        <svg class="w-3 h-3 me-2" aria-hidden="true"
+                                            xmlns="http://www.w3.org/2000/svg" fill="currentColor"
+                                            viewBox="0 0 20 20">
+                                            <path d="M14.707 7.793a1 1 0 0 0-1.414 0L11 10.086V1.5a1 1 0 0 0-2 0v8.586L6.707 7.793a1 1 0 1 0-1.414 1.414l4 4a1 1 0 0 0 1.416 0l4-4a1 1 0 0 0-.002-1.414Z" />
+                                            <path d="M18 12h-2.55l-2.975 2.975a3.5 3.5 0 0 1-4.95 0L4.55 12H2a2 2 0 0 0-2 2v4a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2v-4a2 2 0 0 0-2-2Zm-3 5a1 1 0 1 1 0-2 1 1 0 0 1 0 2Z" />
+                                        </svg>Delete
+                                    </button>
+                                </form>
+                            </li>
+                        </ul>
+                    </div>
+                </div>
+            </td>
+        </tr>
+    `;
 }
 
+/**
+ * Creates a priority badge based on the priority number
+ * @param {number} priority The priority of the task
+ * @return {string} A priority badge as a string
+ */
+function createPriorityBadge(priority) {
+    let badgeClass = '';
+    let badgeText = '';
+
+    switch (priority) {
+        case 1:
+            badgeClass = 'bg-blue-100 text-blue-800 dark:bg-blue-900 dark:text-blue-300';
+            badgeText = 'Low';
+            break;
+        case 2:
+            badgeClass = 'bg-yellow-100 text-yellow-800 dark:bg-yellow-900 dark:text-yellow-300';
+            badgeText = 'Medium';
+            break;
+        case 3:
+            badgeClass = 'bg-red-100 text-red-800 dark:bg-red-900 dark:text-red-300';
+            badgeText = 'High';
+            break;
+    }
 
-if (document.getElementById("pie-chart") && typeof ApexCharts !== 'undefined') {
-    const chart = new ApexCharts(document.getElementById("pie-chart"), getPieCharOptions());
-    chart.render();
+    return `<span class="text-xs font-medium me-2 px-2.5 py-0.5 rounded ${badgeClass}">${badgeText}</span>`;
 }
 
-console.log(document.getElementById("column-chart"));
+/**
+ * Deletes a task
+ * @param {Event} event The event that triggered the deletion
+ */
+function deleteTask(event) {
+    event.preventDefault();
+    const url = $(event.target).closest('form').attr('action');
+    console.debug(`Deleting task at url: ${url}`);
+    
+    const id = $(event.target).parents('form').attr('data-task-id');
+    $.ajax({
+        url: url,
+        type: 'DELETE',
+        headers: {
+            'X-CSRFToken':csrfToken 
+        },
+        data: { id: id },
```

Comment:
I don't think you need `data` here
```suggestion
```

Reply:
not sure why its here to be honest but I have updated it on the next upcoming update